controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1061,15 +1061,18 @@ func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner,
 	}
 
 	prepareData := provisioner.PrepareData{
-		TargetRAIDConfig: newStatus.Provisioning.RAID.DeepCopy(),
-		ActualRAIDConfig: info.host.Status.Provisioning.RAID.DeepCopy(),
-		RootDeviceHints:  newStatus.Provisioning.RootDeviceHints.DeepCopy(),
-		FirmwareConfig:   newStatus.Provisioning.Firmware.DeepCopy(),
+		TargetRAIDConfig:    newStatus.Provisioning.RAID.DeepCopy(),
+		ActualRAIDConfig:    info.host.Status.Provisioning.RAID.DeepCopy(),
+		RootDeviceHints:     newStatus.Provisioning.RootDeviceHints.DeepCopy(),
+		FirmwareConfig:      newStatus.Provisioning.Firmware.DeepCopy(),
+		ManualCleanFinished: info.host.Status.Provisioning.ManualCleanFinished,
 	}
 	// When manual cleaning fails, we think that the existing RAID configuration
 	// is invalid and needs to be reconfigured.
 	if info.host.Status.ErrorType == metal3v1alpha1.PreparationError {
-		prepareData.ActualRAIDConfig = nil
+		if !prepareData.ManualCleanFinished {
```

Comment:
If this is the only place where we need to know whether manual cleaning had finished in a previous reconcile, we could potentially store that info in the ErrorReason, or create a second kind of PreparationError (ConfigurationError or something) to signal that the failure occurred during manual cleaning rather than automated cleaning.

I'm leaning toward the latter.

Reply:
Thanks for your comments, we will rewrite the PR with your suggested approach.

Reply:
@zaneb Thanks for you review, for some reason I took over this PR from @hs0210.

I found an easier way to do this by not clearing `host provisioning settings` when auto clean failure.
In this way we can avoid using additional flags or adding new `PreparationError`. WDYT?

Reply:
I tested with iRMC and manually let the auto clean fail, following is a part of BMO's log:
```log
2023-02-07T10:00:19.216Z        INFO    provisioner.ironic      current provision state {"host": "openshift-machine-api~worker-test", "lastError": "", "current": "clean failed", "target": "available"}
......
......
2023-02-07T10:00:43.435Z        DEBUG   provisioner.ironic      found existing node by ID       {"host": "openshift-machine-api~worker-test"}
2023-02-07T10:00:43.435Z        ERROR   provisioner.ironic      auto clean failed       {"host": "openshift-machine-api~worker-test", "error": "By request, the clean operation was aborted"}
2023-02-07T10:00:43.435Z        INFO    provisioner.ironic      changing provisioning state     {"host": "openshift-machine-api~worker-test", "current": "clean failed", "existing target": "available", "new target": "manage"}
......
......
2023-02-07T10:00:53.555Z        INFO    provisioner.ironic      changing provisioning state     {"host": "openshift-machine-api~worker-test", "current": "manageable", "existing target": "", "new target": "provide"}
......
......
2023-02-07T10:01:13.724Z        INFO    provisioner.ironic      current provision state {"host": "openshift-machine-api~worker-test", "lastError": "", "current": "cleaning", "target": "available"}
{"level":"info","ts":"2023-02-07T10:01:13.724Z","logger":"controllers.BareMetalHost","msg":"using hardware profile","baremetalhost":"openshift-machine-api/worker-test","provisioningState":"preparing","profile":"unknown"}
......
......
2023-02-07T10:15:07.917Z        DEBUG   provisioner.ironic      found existing node by ID       {"host": "openshift-machine-api~worker-test"}
{"level":"info","ts":"2023-02-07T10:15:07.917Z","logger":"controllers.BareMetalHost","msg":"done","baremetalhost":"openshift-machine-api/worker-test","provisioningState":"available","requeue":true,"after":60}
```

It can be confirmed from the log that only `auto clean` will be restarted after `clean failed`.